File: flickr_interestingness.py
import xml.etree.ElementTree as ET
import urllib2
import re
import os

# remove current images
working_dir = os.path.dirname(os.path.realpath(__file__))

for root, dirs, files in os.walk(working_dir, topdown=False):
  for file in files:
    if os.path.splitext(file)[1].lower() in ('.jpg', '.jpeg'):
      # for 500px compatibility. 500px file does not contain '_'
      if '_' in file:
        full_path = os.path.join(root, file)
        # print os.path.join(root, file)
        os.remove(full_path)

# fetch rss
rss_url = 'http://feeds.feedburner.com/LargeFlickrInterestingnessFeedForWallpaper'
res = urllib2.urlopen(rss_url)
rss = res.read()

# fetch images
root = ET.XML(rss)

for item in root[0].iter('item'):
  link = item.find('link').text
  filename = re.split('\/', link)[-1]
  jpg = urllib2.urlopen(link)
  fh = open(working_dir + "/" + filename, 'w')
  fh.write(jpg.read())
  fh.close()

File: flickr_interestingness.rb
require 'rss'
require 'yaml'
require 'json'
require 'open-uri'

# read config file
CONFIG_PATH = './config.yml'
cfg = YAML.load_file(CONFIG_PATH)
WORKING_DIR = ( cfg['working_dir'] && Dir.exists?(cfg['working_dir']) ? cfg['working_dir'] : Dir.pwd )

api_fi_url = "https://api.flickr.com/services/rest/?" +
             "method=flickr.interestingness.getList&" +
             "format=json&" +
             "nojsoncallback=1&" +
             "api_key=" + cfg['flickr']['key']

# remove current images
## flickr photos contain '_'
Dir[WORKING_DIR + "/*.jpg"].select {|x| x =~ /_/}.each do |f|
  File.unlink(f)
end

# read RSS and fetch images
#open(RSS_URL) do |rss|
rsp = JSON.load(open(api_fi_url))
if rsp['stat'] == 'ok'
  rsp['photos']['photo'].each do |photo|
    api_psize_url = "https://api.flickr.com/services/rest/?" +
                    "method=flickr.photos.getSizes&" +
                    "format=json&" +
                    "nojsoncallback=1&" +
                    "api_key=" + cfg['flickr']['key'] + "&" +
                    "photo_id=" + photo['id']
    rsp_sizes = JSON.load(open(api_psize_url))
    if rsp_sizes['stat'] == 'ok'
      largest = rsp_sizes['sizes']['size'][-1]
      if [largest['width'].to_i, largest['height'].to_i].max >= 1920
        filename = largest['source'].split('/')[-1]
        file_path = WORKING_DIR + "/" + filename

        # fetch image and write
        File.open(file_path, 'w') do |f|
          f.write open(largest['source']).read
        end
      end
    end
  end
end